### ISSUE TITLE
PHP8.1 Deprecations

### ISSUE THREAD

**Issue by oleibman** [PHPOffice/PhpSpreadsheet#1843]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
All unit tests should pass under PHP8.1.

### What is the current behavior?
Over the last day or so, there appears to have been an update which deprecates passing null to a number of builtin functions which are expecting strings. I noticed this pushing an update to PR #1828. Deprecations in the following were addressed as part of subsequent pushes to that PR, using null coalescing on a single statement in each:
- Shared/Font
- Shared/XMLWriter
- Style/Color
- Writer/HTML
- Shared/StringHelper

Additional errors ascribed to Cell/Cell seem to have been addressed with several null coalescing operations on all calls to strcmp in Calculation/Calculation, a mysterious but effective solution. These changes seem to have eliminated all the problems in PhpSpreadsheet, except ...

PHP8.1 tests still failed due to a similar problem in vendor/mpdf/mpdf/src/Mpdf. That's clearly out of scope for my PR.

Perhaps someone needs to mark the 8.1 test as "allowed to fail" until Mpdf has a chance to address that, as well as to monitor other similar problems for a bit.

### What are the steps to reproduce?

Run PhpUnit test suite.

### Which versions of PhpSpreadsheet and PHP are affected?
PHP8.1 is the only version affected, regardless of PhpSpreadsheet version.

**Comment by MarkBaker** [comment]
The good news is that we don't need to worry about these till closer to the PHP8.1 release at the end of the year: the bad news is that we need to figure out how to handle an allowed failure in the github actions for PHP 8.1, because we'll almost certainly hit other 8.1 issues

**Comment by MarkBaker** [comment]
I've modified the github actions to provide allow failure for PHP8.1 unit tests. Github actions don't actually provide this functionality (despite a lot of requests for it), so I'm treating php-version 8.1 as "experimental", and modifying any failed status against "experimental" to be over-ridden so the run will continue for all the other checks.

This means that PHP 8.1 Tests show as passed despite any errors or failures, and it requires checking the actual output from the run to see what the rea result is; but I can live with that until github provides functionality for a proper allow_failure option.

**Comment by oleibman** [comment]
Thanks - works as advertised. I have successfully pushed my PR, despite the expected PHP8.1 errors. I also reported the problem to Mpdf.

**Comment by MarkBaker** [comment]
The problems that are highlighted in the 8.1 test run can now be addressed gradually over time; but an 8.1 fail isn't blocking anything else now, and the expanded build result for 8.1 does still give us visibility of the problems
